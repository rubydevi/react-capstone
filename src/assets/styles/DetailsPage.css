.pokemon-details {
  display: flex;
  flex-direction: column;
  height: 100vh;
}

.pokemon-header {
  display: flex;
  align-items: center;
  justify-content: space-between;
  gap: 10px;
  width: 100%;
  padding: 3%;
  background-color: #fb5193;
}

h2 {
  background-color: #cf4278;
  color: #fff;
  padding: 1%;
  text-transform: uppercase;
  font-size: 0.8em;
  padding-left: 10%;
}

.pokemon-image {
  width: 40%;
}

.pokemon-name {
  font-size: 1.5em;
  color: #fff;
}

.pokemon-profile {
  list-style: none;
  padding: 0;
}

.pokemon-profile li {
  display: flex;
  padding: 8px;
  color: #fff;
  height: 70%;
  font-size: 1.2em;
  font-weight: 600;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
}

/* Apply alternating background colors to list items */
.pokemon-profile li:nth-child(odd) {
  background-color: #f65090;
}

.pokemon-profile li:nth-child(even) {
  background-color: #cf4278;
}
