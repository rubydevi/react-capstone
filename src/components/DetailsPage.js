import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useParams } from 'react-router-dom';
import { fetchPokemonDetails } from '../redux/pokemon/pokemonSlice';
import '../assets/styles/DetailsPage.css';

const DetailsPage = () => {
  const { pokemonName } = useParams();
  const selectedPokemon = useSelector((state) => state.pokemons.selectedPokemon);
  const pokemons = useSelector((state) => state.pokemons.pokemons);
  const dispatch = useDispatch();

  useEffect(() => {
    if (selectedPokemon && selectedPokemon.name && selectedPokemon.value) {
      dispatch(fetchPokemonDetails(selectedPokemon.name));
    }
  }, [dispatch, pokemonName, selectedPokemon]);

  if (!selectedPokemon) {
    return (
      <div className="pokemon-details load">
        <div className="loader" />
      </div>
    );
  }

  // Find the selected Pokemon's details from the pokemons array
  const selectedPokemonObject = pokemons.find((pokemon) => pokemon.name === selectedPokemon.name);

  // Access the details from the selected Pokemon's object
  const selectedPokemonDetails = selectedPokemonObject ? selectedPokemonObject.details : null;

  if (!selectedPokemonDetails) {
    return (
      <div className="pokemon-details load">
        <div className="loader" />
      </div>
    );
  }

  const { name, value } = selectedPokemon;
  const defaultImage = selectedPokemonDetails.sprites.other.dream_world.front_default;

  return (
    <div className="pokemon-details">
      <div className="pokemon-header">
        <img className="pokemon-image" src={defaultImage} alt={name} />
        <h1 className="pokemon-name">{name.toUpperCase()}</h1>
      </div>
      <h2>
        {name}
        &apos;s Details
      </h2>
      <ul className="pokemon-profile">
        <li>
          <span>Pokemon Number</span>
          <span>{value}</span>
        </li>
        <li>
          <span>Base Experience</span>
          <span>{selectedPokemonDetails.base_experience}</span>
        </li>
        <li>
          <span>Weight</span>
          <span>{selectedPokemonDetails.weight}</span>
        </li>
        <li>
          <span>Height</span>
          <span>{selectedPokemonDetails.height}</span>
        </li>
      </ul>
    </div>
  );
};

export default DetailsPage;
