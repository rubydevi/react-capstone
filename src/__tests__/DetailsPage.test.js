import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import DetailsPage from '../components/DetailsPage';

const mockStore = configureStore([]);

test('DetailsPage component snapshot', () => {
  const initialState = {
    pokemons: {
      selectedPokemon: {
        name: 'Pikachu',
        value: 25,
      },
      pokemons: [
        {
          name: 'Pikachu',
          details: {
            sprites: {
              other: {
                dream_world: {
                  front_default: 'path/to/pikachu-image.png',
                },
              },
            },
            base_experience: 112,
            weight: 60,
            height: 4,
          },
        },
      ],
    },
  };
  const store = mockStore(initialState);

  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <DetailsPage />
        </Router>
      </Provider>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
