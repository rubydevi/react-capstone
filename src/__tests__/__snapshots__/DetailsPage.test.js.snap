// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsPage component snapshot 1`] = `
<div
  className="pokemon-details"
>
  <div
    className="pokemon-header"
  >
    <img
      alt="Pikachu"
      className="pokemon-image"
      src="path/to/pikachu-image.png"
    />
    <h1
      className="pokemon-name"
    >
      PIKACHU
    </h1>
  </div>
  <h2>
    Pikachu
    's Details
  </h2>
  <ul
    className="pokemon-profile"
  >
    <li>
      <span>
        Pokemon Number
      </span>
      <span>
        25
      </span>
    </li>
    <li>
      <span>
        Base Experience
      </span>
      <span>
        112
      </span>
    </li>
    <li>
      <span>
        Weight
      </span>
      <span>
        60
      </span>
    </li>
    <li>
      <span>
        Height
      </span>
      <span>
        4
      </span>
    </li>
  </ul>
</div>
`;
